server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -161,23 +161,23 @@ private List<ContextResource> getDataSourceResources(ContextProperties props) th
                     dataSourceResource.setProperty("username", props.getUsername().get(i));
 
                     String maxTotal = props.getMaxTotal().get(i);
```

Comment:
Based on the error Alan shared, `props.getMaxTotal()` is returning null:

```
org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is java.lang.NullPointerException: Cannot invoke "java.util.List.get(int)" because the return value of "org.labkey.embedded.LabKeyServer$ContextProperties.getMaxTotal()" is null
```

I think we need null checks on the list itself, and checks on its length too.

Reply:
I added a null check for the list & caught the IndexExceptions and return the default value for both. Could possibly use a Map instead of a List which would allow key checking, but not sure of any secondary ramifications of the type change.